src/analyzer/analyzersilence.cpp

```diff
@@ -24,22 +24,22 @@ bool hasOutroCueEnd(const Cue& outroCue) {
 }
 
 bool needsIntroCueStart(const Cue& introCue) {
-    return introCue.getSource() != Cue::MANUAL &&
+    return introCue.getSource() != Cue::Source::Manual &&
             !hasIntroCueStart(introCue);
 }
 
 bool needsOutroCueEnd(const Cue& outroCue) {
-    return outroCue.getSource() != Cue::MANUAL &&
+    return outroCue.getSource() != Cue::Source::Manual &&
             !hasOutroCueEnd(outroCue);
 }
 
-bool shouldAnalyze(TrackPointer tio) {
-    CuePointer pIntroCue = tio->findCueByType(Cue::INTRO);
-    if (!pIntroCue) {
-        return true;
-    }
-    CuePointer pOutroCue = tio->findCueByType(Cue::OUTRO);
-    if (!pOutroCue) {
+bool shouldAnalyze(TrackPointer pTrack) {
+    CuePointer pIntroCue = pTrack->findCueByType(Cue::Type::Intro);
+    CuePointer pOutroCue = pTrack->findCueByType(Cue::Type::Outro);
+    CuePointer pFirstSound = pTrack->findCueByType(Cue::Type::FirstSound);
+    CuePointer pLastSound = pTrack->findCueByType(Cue::Type::LastSound);
+
+    if (!pIntroCue || !pOutroCue || !pFirstSound || !pLastSound) {
         return true;
     }
     return needsIntroCueStart(*pIntroCue) || needsOutroCueEnd(*pOutroCue);
```

Comment:
What do you mean? What specifically do you think should be changed?

Reply:
This is also what came to my mind. But we also need to consider pIntroCue and pOutro cue here! Otherwise they won't be updated as desired.

- If IntroStart or OutroEnd are missing we should set them according to FirstSound or LastSound respectively. This is done in storeResults() **after** the analysis has finished.
- It is always save to recalculate FirstSound and LastSound, even if they already exist.

Corollary: Even if both FirstSound and LastSound exist we need to re-analyze if we want to set IntroStart or OutroEnd. Otherwise storeReults() is not invoked.

Reply:
So it is fine, as it is?

Reply:
I think so. We might even consider the main cue point again here, now that we have those implicit markers that can safely be reanalyzed. That would only be consequent.